### ISSUE TITLE
Allow services to emulate user (send with username)

### ISSUE THREAD

**Issue by tmandry** [gitterHQ/gitter#1076]
It would be great if the API allowed bots to specify a username with each request. This would make it possible to write a [bridge to a Matrix](https://matrix.org/jira/browse/BOTS-133) or IRC channel, for instance, with matching usernames.

Related: #662

**Comment by MadLittleMods** [comment]
Related to https://github.com/gitterHQ/gitter/issues/158 ?

**Comment by tmandry** [comment]
No, that's for the bridge which allows people to use Gitter from an IRC client. I'm talking about a two-way bridge between a real IRC channel (on, say, Freenode) and a Gitter room.

**Comment by MadLittleMods** [comment]
Duplicate of https://github.com/gitterHQ/gitter/issues/662

**Comment by tmandry** [comment]
Hi, I don't really consider this a duplicate because it

- has a more general use case than #662
- is less work to implement than #662 

as an example of a more general use case, in one project I worked on we would post user messages to a slack channel as the user's name and with the user's picture. It didn't allow interaction but was a more succinct way to display the user activity on our product.

**Comment by MadLittleMods** [comment]
To be clear, are you wanting to emulate a message from an existing user?

How do we avoid spoofed badness? Use a username that is prefixed with the service name `irc-sync~tmandry`, etc?

**Comment by tmandry** [comment]
It wouldn't be tied to a particular github or gitter user, just any arbitrary username. You could flag the messages as coming from a bot and put that in smaller, lighter text next to the name. The prefix approach (or giving all messages from the bot the same profile picture) could work too.





On Thu, Jan 21, 2016 at 9:01 AM -0800, "Eric Eastwood" <notifications@github.com> wrote:












To be clear, are you wanting to emulate a message from an existing user?



How do we avoid spoofed badness? Use a username that is prefixed with the service name irc-sync~tmandry, etc?

**Comment by finnp** [comment]
Yeah it would be like Slack bot users (https://api.slack.com/bot-users).  The name would be flagged with something like "Bot". Some kind of prefix including characters which are not allowed for github user names would be ok too.

**Comment by leonerd** [comment]
:+1:

**Comment by leonerd** [comment]
You might further be interested to see my initial hack at an implementation of a Matrix <-> Gitter bridge script; https://github.com/matrix-org/matrix-appservice-gitter
On the Matrix side it already lazily provisions a ghost user to represent messages from Gitter, but it can't (yet) do the same on the Gitter side for messages from Matrix. If this feature were added it would be a nice two-way bridge setup.

**Comment by stevezau** [comment]
+1

**Comment by MatthewSteen** [comment]
+1

**Comment by last-khajiit** [comment]
+1

**Comment by opjlmi** [comment]
:+1:
